api/webrtc/webrtc.go

```diff
@@ -99,8 +101,10 @@ func (c *Peer) Offer(label string) error {
 	if err != nil {
 		return err
 	}
+
 	dc.OnOpen(func() {
 		log.Debugf("Data channel %v has been opened\n", dc.Label())
+		c.OnOfferConnected <- struct{}{}
```

Comment:
need to use default to prevent blocking since Offer() can be called multiple times

Reply:
Yes, modified.